### ISSUE TITLE
Can create invalid geometries with node merge.

### ISSUE THREAD

**Issue by atiannicelli** [openstreetmap/iD#9328]
I've been attempting to figure out ways that mappers could have produced a few different types of invalid areas. and this is one such way that allows a user to commit an invalid way with no warnings. 

### URL

I don't have a good example on the map of this as I have been trying to clean them up, but here is an example of an invalid way that was created by iD. 
https://www.openstreetmap.org/way/545586121

### How to reproduce the issue?

1. Click on "area"
2. create a 5 node area on the map with one node in the middle of the other four nodes.
3. select the two nodes that are next to the 5th node.
4) type "c" or Merge the two nodes.

At this point you will have a triangle with a line.
The change file shows a five node area with one node referenced three times.

### Screenshot(s) or anything else?

Before merging:
![Screen Shot 2022-10-06 at 3 00 16 PM](https://user-images.githubusercontent.com/5851115/194396643-7fe156c7-5b6f-4492-bf91-1cefdbfbb147.png)

After merging nodes:
![Screen Shot 2022-10-06 at 3 04 16 PM](https://user-images.githubusercontent.com/5851115/194397410-891b803b-f583-46d4-9852-70fde87d080e.png)

```
<osmChange version="0.6" generator="iD">
    <create>
        <node id="-1" lon="116.36608058169551" lat="39.98412126265558" version="0"/>
        <node id="-2" lon="116.36602189887817" lat="39.984038260600855" version="0"/>
        <node id="-3" lon="116.36617426779885" lat="39.98403553546644" version="0"/>
        <node id="-5" lon="116.36626760913097" lat="39.98411368268282" version="0"/>
        <way id="-1" version="0"><nd ref="-1"/><nd ref="-2"/><nd ref="-3"/><nd ref="-1"/><nd ref="-5"/><nd ref="-1"/><tag k="area" v="yes"/></way>
    </create>
    <modify/>
    <delete if-unused="true"/>
</osmChange>
```

### Which deployed environments do you see the issue in?

Released version at openstreetmap.org/edit, RapiD version at mapwith.ai/rapid

### What version numbers does this issue effect?

2.22.0

### Which browsers are you seeing this problem on?

Chrome

**Comment by jguthrie100** [comment]
I'd be happy to work on this.

@tyrasd  - If two points in a polygon are merged, does it make sense to just delete every node between those two points? To me that seems like its how it should happen. Everything between those two points should just get zapped..

**Comment by atiannicelli** [comment]
Makes sense to me. And they always have the Ctrl-Z if it does something other than what they expected. 

The other way to go would be to just refuse to merge two nodes that are members of the same way or relation that are not adjacent to each other.

**Comment by tyrasd** [comment]
Confirming the bug: while the drag-and-drop vertex snapping prevents one from creating such invalid geometries, this is not yet implemented for the combine operation.

> If two points in a polygon are merged, does it make sense to just delete every node between

I would say that it would be better to just not allow the merge operation for vertices in such cases.

**Comment by k-yle** [comment]
if anyone wants to attempt this (and possibly #11726 at the same time), adding these tests cases would be a good place to start: https://github.com/openstreetmap/iD/pull/11734#pullrequestreview-3630412351
